doc/design/status_port.md

```diff
@@ -0,0 +1,453 @@
+
+            callable(*(m_statusPortDataPtr->chunks[currentReadPosition].data));
+
+            // Re-call the callable if the world changed in the meantime
+        } while (currentTransaction != m_statusPortDataPtr->latestTransaction.load(std::memory_order_acquire));
+    }
+
+  private:
+    StatusPortData<T>* m_statusPortDataPtr;
+    /// @todo #982 add getMembers() when integrating into RouDi infrastructure
+};
+
+template <typename T>
+class StatusPortWriter
+{
+  public:
+    StatusPortWriter(cxx::not_null<StatusPortData<T>*> statusPortDataPtr) noexcept
+        : m_statusPortDataPtr(statusPortDataPtr)
+    /// @todo #982 Replace once the RouDi infrastructure is ready
+    // m_statusPortDataPtr(iox::runtime::PoshRuntime::getInstance().getMiddlewareStatusPort(sizeof(T)))
+    {
+    }
+
+    StatusPortWriter(StatusPortWriter&& rhs) = delete;
+    StatusPortWriter& operator=(StatusPortWriter&& rhs) = delete;
+    StatusPortWriter(const StatusPortWriter&) = delete;
+    StatusPortWriter& operator=(const StatusPortWriter&) = delete;
+
+    void store(cxx::function_ref<void(T&)> callable) noexcept
+    {
+        // Against what do we have to protect us in store?
+        // Against nothing we are the boss and the single entitiy changing latestTransaction, hence no CAS is needed
+
+        // Get current world view
+        auto currentTransaction = m_statusPortDataPtr->latestTransaction.load(std::memory_order_relaxed);
+        // Get the readPosition and toggle it to get write position
+        auto currentWritePosition =
+            1 xor static_cast<std::underlying_type<ActiveChunk>::type>(currentTransaction.activeChunk);
+
+        /// @todo #982 Later we'll directly pass the dereferenced raw shared memory pointer
+        // callable(*(m_statusPortDataPtr->chunks[currentWritePosition].data));
+
+        // Update our new world view, it can't fail because we're the only writer
+        T valueToStore;
+        callable(valueToStore);
+        m_statusPortDataPtr->chunks[currentWritePosition].data.emplace(valueToStore);
+
+        Transaction newTransaction{static_cast<ActiveChunk>(currentWritePosition), ++currentTransaction.abaCounter};
+        m_statusPortDataPtr->latestTransaction.store(newTransaction, std::memory_order_release);
+        // Store operation aka world view is now observable for all readers
+    }
+
+  private:
+    StatusPortData<T>* m_statusPortDataPtr;
+    /// @todo #982 add getMembers() when integrating into RouDi infrastructure
+};
+
+#endif // IOX_POSH_POPO_STATUS_PORT_HPP
+
+} // namespace popo
+} // namespace iox
+```
+
+## Open issues
+
+* Naming
+    * `StatusPort{Writer,Reader}` or just `StatusReader` and `StatusWriter`?
+* Does `copyTake()` make sense as an additional contract with the user?
+    * It would be possible that the user provides an arbitrary lambda, that does
+    not need to be written with care
```

Comment:
I am also not sure that the read/take algorithm does not have a problem where we may read corrupt data without detecting it. I will have to check some material and think about it and will get back to this later.